src/main/java/seedu/address/model/activity/Amount.java

```diff
@@ -0,0 +1,43 @@
+package seedu.address.model.activity;
+
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+/**
+ * Represents an expense Amount in an Expense.
+ * Guarantees: immutable; is valid as declared in {@link #isValidAmount(double)}
+ */
+public class Amount {
+    public static final String MESSAGE_CONSTRAINTS =
+            "The amount of money spent can only be a positive number";
+
+    public final double value;
+
+    /**
+     * Constructs an {@code Amount}.
+     *
+     * @param amount A valid amount of money spent.
+     */
+    public Amount(double amount) {
+        checkArgument(isValidAmount(amount), MESSAGE_CONSTRAINTS);
+        value = amount;
+    }
+
+    /**
+     * Returns true if a given string is a valid name.
+     */
+    public static boolean isValidAmount(double test) {
+        return test > 0;
+    }
+
+    @Override
+    public String toString() {
+        return String.format("%f", value);
+    }
+
+    @Override
+    public boolean equals(Object other) {
+        return other == this // short circuit if same object
```

Comment:
Use guard conditions,
```
if (other == this) return true;
if (other instanceof Amount) 
    return value == ((Amount) other).value;
return false;

Reply:
adjusted, although the existing AB3 code base actually uses the one without guard conditions...